audio-worklet/design-pattern/awn/audio-worklet-processor.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * A simple demonstration of WASM-powered AudioWorkletProcessor.
+ *
+ * @class WASMAudioWorkletProcessor
+ * @extends AudioWorkletProcessor
+ */
+class WASMAudioWorkletProcessor extends AudioWorkletProcessor {
+
+  constructor() {
+    super();
+
+    // Allocate the buffer for the heap access. Start with stereo, but it can
+    // be expanded up to 32 channels.
+    this._heapInputBuffer =
+        new HeapAudioBuffer(WA2.RENDER_QUANTUM_FRAMES, 2, 32);
+    this._heapOutputBuffer =
+        new HeapAudioBuffer(WA2.RENDER_QUANTUM_FRAMES, 2, 32);
+
+    this._kernel = new Module.AudioWorkletProcessorKernel();
+  }
+
+  process(inputs, outputs, parameters) {
+    // Use the 1st input and output only to make the example simpler. |input|
+    // and |output| here have the similar structure with the AudioBuffer
+    // interface. (i.e. An array of Float32Array)
+    let input = inputs[0];
+    let output = outputs[0];
+
+    // The channel count of the pipeline in the node is static within a given
+    // render quantum. Also identical for the input and output.
+    let channelCount = input.length;
+
+    // Prepare the dynamic channel count change.
```

Comment:
It's kind of confusing to say "dynamic" here and line 20 says "static".

Reply:
Revised the comment to clarify.